src/Workspaces/Core/Portable/Execution/PinnedRemotableDataScope.cs

```diff
@@ -63,7 +64,12 @@ public RemotableData GetRemotableData(Checksum checksum, CancellationToken cance
 
         public void Dispose()
         {
-            _storages.UnregisterSnapshot(this);
+            if (!_disposed)
```

Comment:
Can you comment how double disposable is currently possible.  But link to issue that tracks just cleaning all this up.